File: src/visual-graph/builder/return-subgraph-id.test.ts
import { describe, expect, test } from "vitest";

import { returnSubgraphId } from "./return-subgraph-id.js";

describe("returnSubgraphId", () => {
  test.each([
    { varId: "v1", containerKey: "10-20", expected: "s_return_v1_10_20" },
    {
      varId: "v.1",
      containerKey: "implicit",
      expected: "s_return_v_1_implicit",
    },
    { varId: "", containerKey: "", expected: "s_return__" },
    { varId: "owner-x", containerKey: "5-9", expected: "s_return_owner_x_5_9" },
  ])(
    "returnSubgraphId($varId, $containerKey) = $expected",
    ({ varId, containerKey, expected }) => {
      expect(returnSubgraphId(varId, containerKey)).toBe(expected);
    },
  );
});

File: crates/unsnarl-visual-graph/src/builder/return_subgraph_id_test.rs
//! Sibling tests for [`return_subgraph_id`].

use super::return_subgraph_id;

#[test]
fn alphanumeric_var_and_key_pass_through() {
    assert_eq!(return_subgraph_id("v1", "10-20"), "s_return_v1_10_20");
}

#[test]
fn specials_in_var_id_are_replaced() {
    assert_eq!(
        return_subgraph_id("v.1", "implicit"),
        "s_return_v_1_implicit"
    );
}

#[test]
fn empty_inputs_keep_separators() {
    assert_eq!(return_subgraph_id("", ""), "s_return__");
}

#[test]
fn specials_in_both_arguments_are_replaced() {
    assert_eq!(return_subgraph_id("owner-x", "5-9"), "s_return_owner_x_5_9");
}
